src/soundbase.cpp

```diff
@@ -166,30 +166,38 @@ QString CSoundBase::SetDev ( const QString strDevName )
         if ( !vsErrorList.isEmpty() )
         {
             // create error message with all details
-            QString sErrorMessage = "<b>" + QString ( tr ( "No usable %1 audio device found." ) ).arg ( strSystemDriverTechniqueName ) + "</b><br>";
+            QString sErrorMessage = "<b>";
             if ( lNumDevs > 1 )
             {
                 // only output multiple messages if more then one are necessary.
-                sErrorMessage += "<br>" + tr ( "These are all the available drivers with error messages:" ) + "<ul>";
+                sErrorMessage += tr ( "Sound driver errors (using %1) found:</b><br>"
+                                      "None of your sound devices are compatible in their current state." )
+                                     .arg ( strSystemDriverTechniqueName );
+
                 for ( int i = 0; i < lNumDevs; i++ )
                 {
-                    sErrorMessage += "<li><b>" + GetDeviceName ( i ) + "</b>: " + vsErrorList[i] + "</li>";
+                    sErrorMessage += "<b>" + GetDeviceName ( i ) + "</b>: " + vsErrorList[i] + "<br>";
                 }
-                sErrorMessage += "</ul>";
             }
             else
             {
-                 sErrorMessage += "<b>" + GetDeviceName ( 0 ) + "</b>: " + vsErrorList[0] ;
+                sErrorMessage += tr ( "Couldn't initialize audio driver (using %1)</b><br>"
+                                      "Your sound device is not compatible in its current state." )
+                                     .arg ( strSystemDriverTechniqueName );
+                sErrorMessage += "<b>" + GetDeviceName ( 0 ) + "</b>: " + vsErrorList[0];
             }
 
-
 #ifdef _WIN32
             // to be able to access the ASIO driver setup for changing, e.g., the sample rate, we
             // offer the user under Windows that we open the driver setups of all registered
             // ASIO drivers
-            sErrorMessage += "<br/>" + tr ( "You might be able to fix this error by changing settings of your ASIO driver. Do you want to open these settings now?" );
+            sErrorMessage += "<br/>" + tr ( "You may be able to fix errors in the driver settings." );
+            // for Windows use a QMessageBox to show the error
+
+            QMessageBox::information mbWinErrorBox ( nullptr, APP_NAME, sErrorMessage, QMessageBox::Cancel );
+            QPushButton*             btnASIOSettings = mbWinErrorBox.addButton ( tr ( "Open ASIO settings" ), QMessageBox::AcceptRole );
 
-            if ( QMessageBox::Yes == QMessageBox::information ( nullptr, APP_NAME, sErrorMessage, QMessageBox::Yes | QMessageBox::No ) )
+            if ( mbWinErrorBox.clickedButton() == btnASIOSettings )
             {
                 LoadAndInitializeFirstValidDriver ( true );
```

Comment:
`vsErrorList = LoadAndInitializeFirstValidDriver ( true );` and then check the list size.  If it's zero (and I've not checked the code to make sure I'm right here...), `return strReturn` rather than continuing.

Reply:
I think there's more work needed then just that. See the commit I mentioned: Probably it's in a deeper level.